src/raft.rs

```diff
@@ -217,18 +217,19 @@ pub fn vote_resp_msg_type(t: MessageType) -> MessageType {
 impl<T: Storage> Raft<T> {
     /// Creates a new raft for use on the node.
     #[allow(clippy::new_ret_no_self)]
-    pub fn new(c: &Config, store: T, logger: &Logger) -> Result<Self> {
+    pub fn new(mut c: Config, store: T, logger: &Logger) -> Result<Self> {
```

Comment:
Better not change the interface. I don't see any point to make it owned.

Reply:
Reverted. It's because `validate` need a `&mut self`